### ISSUE TITLE
bspwm : hidden dependency

### ISSUE THREAD

**Issue by gpyh** [voidlinux/void-packages#1144]
TL;DR: Bspwm would show a black screen with an underscore in the top left corner and the keyboard wouldn't respond. I installed libEGL (not listed in the dependencies) and everything works fine now. If I remove libEGL, the same problem happens again.

More background: I installed void linux today after getting rid of my arch install (systemd getting on my nerves). Bspwm worked fine on arch but wouldn't work on void. I installed i3 to see if it was a problem with my machine. Turns out i3 worked like a charm. But so did bspwm ! I then proceeded to remove the dependencies of i3 one by one until bspwm would cease to work. The package libEGL seems to be the culprit. It is not listed in the dependencies of bspwm in xbps, but it is not listed in the ABS either (see https://www.archlinux.org/packages/community/x86_64/bspwm/ ). I don't know if the problem comes from upstream. I'll file this bug report to them. I didn't try to reproduce the bug on a barebone arch install either (sorry I'm lazy). Adding libEGL as a dependency would be a workaround. It is a very small library, but asking upstream should be wiser.

EDIT: Upstream issue : https://github.com/baskerville/bspwm/issues/221

**Comment by xtraeme** [comment]
I believe libEGL is unrelated to your issue :-)

It's just a coincidence that it magically makes bspwm work.

```
[juan@leysa void-packages]$ xbps-query --fulldeptree -Rx bspwm
glibc-2.21_3
libXdmcp-1.1.1_4
libXau-1.0.8_1
libxcb-1.11_2
xcb-util-wm-0.4.1_2
xcb-util-0.4.0_1
[juan@leysa void-packages]$ xbps-query --fulldeptree -Rx libEGL
glibc-2.21_3
libpciaccess-0.13.3_1
libdrm-2.4.59_1
eudev-libudev-2.1.1_6
libglapi-10.5.0_2
libffi-3.2.1_1
wayland-1.7.0_1
expat-2.1.0_4
libgbm-10.5.0_2
libXdmcp-1.1.1_4
libXau-1.0.8_1
libxcb-1.11_2
libX11-1.6.3_1
[juan@leysa void-packages]$
```

bspwm does not need/depend on GL/GLES at all.

**Comment by gpyh** [comment]
That's also what I figured out.
Weird stuff happens I guess. I might investigate further this weekend.
Thanks !

**Comment by dominikh** [comment]
@gpyh did you do any more research or should we time-out this issue?

**Comment by gpyh** [comment]
Sadly I didn't.

**Comment by xtraeme** [comment]
This is probably a missing X driver misconfiguration or unsatisfied dependency of the DDX.

Cannot be reproduced, and this looks like to me PEBKAC.

**Comment by svenper** [comment]
I have only a file for the trackpad in `/etc/X11/xorg.conf.d`, but nothing else, and I have this problem for `dwm`. I have only two drivers for input and using modesetting for graphics. Maybe it's a problem for Xorg in general?

```
# /etc/X11/xorg.conf.d/50-mtrack.conf
Section "InputClass"
        Identifier "Multi-touch trackpad"
        Driver "mtrack"
        MatchDevicePath "/dev/input/event7"
EndSection

Section "InputClass"
        Identifier "Ignore duplicate mouse events"
        MatchOS "Linux"
        MatchDevicePath "/dev/input/mouse*"
        Option "Ignore" "on"
EndSection
```

```
$ xbps-query -s xf86-
[*] xf86-input-evdev-2.10.5_2 Xorg event device input driver
[*] xf86-input-mtrack-0.4.1_1 Xorg Multitouch Trackpad Driver
```

My guess is that you are using something similarly minimal?
